File: 1572.矩阵对角线元素的和.js
/*
 * @lc app=leetcode.cn id=1572 lang=javascript
 *
 * [1572] 矩阵对角线元素的和
 */

// @lc code=start
/**
 * @param {number[][]} mat
 * @return {number}
 */
var diagonalSum = function(mat) {
    let n = mat.length;
    let res = 0;
    for (let i = 0; i < n; i++) {
        res += mat[i][i];
        res += mat[n - 1 - i][i];
    }
    if (n % 2 === 1) {
        let x = Math.floor(n / 2);
        res -= mat[x][x];
    }
    return res;
};
// @lc code=end

File: 1572.矩阵对角线元素的和.java
/*
 * @lc app=leetcode.cn id=1572 lang=java
 *
 * [1572] 矩阵对角线元素的和
 */

// @lc code=start
class Solution {
    public int diagonalSum(int[][] mat) {
        int res = 0;
        int n = mat.length;
        for (int i = 0; i < n; i++) {
            res += mat[i][i];
            res += mat[n - 1 - i][i];
        }
        if (n % 2 == 1) {
            res -= mat[n / 2][n / 2];
        }
        return res;
    }
}
// @lc code=end

